src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,20 +72,29 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
-    }
+        return value_;
+        }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+    	if(other == this){
+    		return true;
+    	} else if(other instanceof Address){
+    		if(this.getBlock().equals(((Address) other).getBlock())){
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.